core/vnl/vnl_matrix_fixed.h

```diff
@@ -667,16 +667,16 @@ class VNL_EXPORT vnl_matrix_fixed
   // for vnl_matrix but not for vnl_matrix_fixed. There is also a
   // conversion operator that should work most of the time.
   // \sa vnl_matrix_ref::non_const
-  inline const vnl_matrix_ref<T> as_ref() const { return vnl_matrix_ref<T>( num_rows, num_cols, const_cast<T*>(data_block()) ); }
```

Comment:
I'm afraid this unusual 'const'-qualification at the top level of the return type does have a meaning here! You see, this is the `const` overload of `as_ref()`. The top-level `const` should protect against modifying the original matrix through the `vnl_matrix_ref`.

Reply:
Reverted:  FYI That is part of another larger issue that I want to keep separate.

the larger issue is my concern about excessive unnecessary implicit type conversions.